File: Lab 10/t2.go
package main

import (
  "fmt"
  "os"
  "bufio"
  "strings"
  "strconv"
)

func main() {
  fmt.Println("Please input a decimal number:")
  line, e := bufio.NewReader(os.Stdin).ReadString('\n')
  if e != nil {
    fmt.Fprintln(os.Stderr, "Read error:", e)
    return
  }
  line = strings.TrimSuffix(line, "\n")

  num, e := strconv.ParseInt(line, 10, 0)
  if e != nil {
    fmt.Fprintln(os.Stderr, "Convert error:", e)
    return
  }

  fmt.Printf("%b\n", num)
}

File: Lab 10/t2.rs
use std::io::{stdin, BufRead};

fn main() {
  println!("Please input a decimal number:");
  let stdin = stdin();
  let num: usize = match stdin.lock().lines().next() {
    Some(r) => match r {
      Ok(r) => match r.parse() {
        Ok(r) => r,
        Err(e) => {
          eprintln!("Parse error: {}", e);
          return;
        }
      },
      Err(e) => {
        eprintln!("Read error: {}", e);
        return;
      }
    },
    None => { return; }
  };

  println!("{:b}", num);
}
